### PR TITLE
Analyze map files to get section sizes of IoT.js build

### PR DESCRIPTION
**Pull request by LaszloLango**
JSRemoteTest-DCO-1.0-Signed-off-by: László Langó llango.u-szeged@partner.samsung.com

### CODE REVIEW

File: API/application/iotjs.py [hunk 1]
```diff
@@ -123,39 +111,63 @@ def build(self, device):
...
+        # 2. Run the buildscript with memstat or Freya
         # For artik053, build IoT.js in os.build.
         if not device.get_type() == 'artik053':
             utils.execute(paths.IOTJS_PATH, 'tools/build.py', build_flags)
 
-        os.build(self, self.buildtype, 'all')
+        os.build(self, self.buildtype, build_flags, 'all')
+
+        # Revert all the memstat patches from the project.
+        if enable_memstat:
```
Comment (bzsolt) [inline API/application/iotjs.py]: This block seems duplicated to [line 173](https://github.com/Samsung/js-remote-test/pull/53/files#diff-43d4ce9fbbfdfaf06d8aeb27c6902667R173).

**Comment by glistening** [comment]
I would like to check its working. However it seems js-remote-test does not work on Ubuntu 14.04. 
Please see #54.

File: API/common/utils.py [hunk 2]
```diff
@@ -210,10 +212,60 @@ def relpath(path, start):
...
+
+    archives = ['libhttpparser.a',
+            'libiotjs.a',
+            'libjerry-core.a',
+            'libjerry-ext.a',
+            'libjerry-port-default.a',
+            'libjerry-port-default-minimal.a',
+            'libtuv.a']
+
+    print("Loading map file '%s'" % mapfile)
```
Comment (rtakacs) [inline API/common/utils.py]: Should we keep this line?

File: API/common/utils.py [hunk 3]
```diff
@@ -210,10 +212,60 @@ def relpath(path, start):
...
+    data = lumpy.load_map_data(mapfile)
+    print("Loaded %d lines" % len(data))
```
Comment (rtakacs) [inline API/common/utils.py]: Ditto.

**Comment by rtakacs** [comment]
I've just checked this PR on stm32f4dis and rpi2 targets and works fine. LGTM.
(I use Ubuntu 16.04.)

File: API/common/utils.py (hunk 2 above)
Comment (LaszloLango) [reply API/common/utils.py]: IMHO it is a useful information, but I can remove it if it does not fit.

**Comment by glistening** [comment]
@LaszloLango Could you share the binary size from this PR? What about generating binary size before running all test cases? Currently I have to wait for completing running test cases to get binary size.

**Comment by LaszloLango** [comment]
RPi2: `"bin": {"total": 311769, "text": 189706, "bss": 265609, "data": 12, "rodata": 122051}`
STM32f4: `"bin": {"total": 284474, "text": 204564, "bss": 83641, "data": 36, "rodata": 79874}`
Artik053: `"bin": {"total": 455285, "text": 300258, "bss": 135084, "data": 76, "rodata": 154951}`

Note: Artik053 is debug build, but the others are release builds.

**Comment by glistening** [comment]
@LaszloLango Thank you for data. @hs0225 will start to support release mode build for artik053.

**Pull request merged**